### PR TITLE
Add doc to Hash select methods - splat variants

### PR DESCRIPTION
**Pull request by jan-zajic**
n/c

### CODE REVIEW

**Comment by jhass** [comment]
Maybe we should just `ditto` them?

**Comment by straight-shoota** [comment]
@jhass But then the code sample would not fit.

In fact, the example for `select(keys : Array | Tuple)` is actually demonstrating `select(*keys)` 🤦‍♂

**Comment by jhass** [comment]
For cases like this I wouldn't mind to have code samples demonstrating all the overloads on the first one and the others just `ditto`ing them.

**Comment by straight-shoota** [comment]
Yeah, that's probably fine.

**Comment by straight-shoota** [comment]
We might even consider some kind of combined presentation for methods with the same (`ditto`) combination. But that's not particularly important.

**Comment by straight-shoota** [comment]
@jan-zajic Would you mind adding a usage example for each `select(keys : Array | Tuple)` and `select!(*keys)` to the dittoed doc string?

**Comment by jan-zajic** [comment]
@straight-shoota like this?

File: src/hash.cr [hunk 1]
```diff
@@ -1418,29 +1418,36 @@ class Hash(K, V)
...
   # {"a" => 1, "b" => 2, "c" => 3, "d" => 4}.select("a", "c") # => {"a" => 1, "c" => 3}
+  # {"a" => 1, "b" => 2, "c" => 3, "d" => 4}.select(["a", "c"]) # => {"a" => 1, "c" => 3}
   # ```
   def select(keys : Array | Tuple)
     hash = {} of K => V
     keys.each { |k| hash[k] = self[k] if has_key?(k) }
     hash
   end
 
+  # ditto
```
Comment (bcardiff) [inline src/hash.cr]:
```suggestion
  # :ditto:
```

File: src/hash.cr [hunk 2]
```diff
@@ -1418,29 +1418,36 @@ class Hash(K, V)
...
+  # h3 = {"a" => 1, "b" => 2, "c" => 3, "d" => 4}.select!(["a", "c"])
+  # h1 == h2 == h3
+  # h1 # => {"a" => 1, "c" => 3}
   # ```
   def select!(keys : Array | Tuple)
     each { |k, v| delete(k) unless keys.includes?(k) }
     self
   end
 
+  # ditto
```
Comment (bcardiff) [inline src/hash.cr]:
```suggestion
  # :ditto:
```

File: src/hash.cr [hunk 3]
```diff
@@ -1418,29 +1418,36 @@ class Hash(K, V)
...
 
   # Removes every element except the given ones.
   #
   # ```
-  # h = {"a" => 1, "b" => 2, "c" => 3, "d" => 4}.select!("a", "c")
-  # h # => {"a" => 1, "c" => 3}
+  # h1 = {"a" => 1, "b" => 2, "c" => 3, "d" => 4}.select!({"a", "c"})
+  # h2 = {"a" => 1, "b" => 2, "c" => 3, "d" => 4}.select!("a", "c")
+  # h3 = {"a" => 1, "b" => 2, "c" => 3, "d" => 4}.select!(["a", "c"])
+  # h1 == h2 == h3
```
Comment (Sija) [inline src/hash.cr]:
```suggestion
  # h1 == h2 == h3 # => true
```

**Comment by straight-shoota** [comment]
Needs a format before being able to merge.

**Comment by jan-zajic** [comment]
@straight-shoota it's done.

**Pull request merged**